crates/agglayer-types/src/aggchain_data/mod.rs

```diff
@@ -0,0 +1,45 @@
+use agglayer_interop_types::aggchain_proof::AggchainData;
+
+mod aggchain_proof;
+mod global;
+mod multisig;
+
+// TODO: better naming for all the structures in this module, including this
+// module name
+pub use crate::aggchain_data::{
+    aggchain_proof::{Ctx as AggchainProofCtx, Payload as AggchainProofPayload},
+    global::{
+        AggchainDataError, Ctx as CertificateAggchainDataCtx, Payload as CertificateAggchainData,
+    },
+    multisig::{Ctx as MultisigCtx, MultisigError, Payload as MultisigPayload},
+    PayloadWithCtx as CertificateAggchainDataWithCtx,
+};
+
+/// Represents the payload from the chain, with the context fetched from the L1.
+#[derive(Clone, Debug)]
+pub struct PayloadWithCtx<Payload, Context>(pub Payload, pub Context);
```

Comment:
Note: At risk of overengineering, you could enforce the correct context type for any given payload by introducing a trait:

```rust
trait PayloadType {
    type Context;
}
```

And using it like this:

```suggestion
pub struct PayloadWithCtx<Payload: PayloadType>(pub Payload, pub Payload::Context);
```